src/main/java/bossman/BossMan.java

```diff
@@ -0,0 +1,166 @@
+        }
+
+        String commandType = parts[0];
+        String commandArgs = parts.length > 1 ? parts[1] : "";
+
+        switch (commandType.toLowerCase()) {
+        case "mark":
+            handleMarkCommand(commandArgs);
+            break;
+
+        case "unmark":
+            handleUnmarkCommand(commandArgs);
+            break;
+
+        case "list":
+            TASK_LIST.printTasks();
+            break;
+
+        case "todo":
+            handleTodoCommand(commandArgs);
+            break;
+
+        case "deadline":
+            handleDeadlineCommand(commandArgs);
+            break;
+
+        case "event":
+            handleEventCommand(commandArgs);
+            break;
+
+        case "bye":
+            break; // No need to print unknown command for 'bye'
+
+        default:
+            System.out.println("Unknown command\n" + SEP);
+        }
+    }
+
+    private String[] parseUserInput(String userInput) {
+        return userInput.trim().split("\\s+", 2);
+    }
+
+    private void handleMarkCommand(String commandArgs) {
+        int number = Integer.parseInt(commandArgs);
+        TASK_LIST.markTask(number);
+    }
+
+    private void handleUnmarkCommand(String commandArgs) throws NumberFormatException{
+        int number = Integer.parseInt(commandArgs);
+        TASK_LIST.unmarkTask(number);
+    }
+
+    private void handleTodoCommand(String commandArgs) throws InvalidTodoCommandException{
+        if(commandArgs.isBlank()) {
+            throw new InvalidTodoCommandException();
+        }
+        Task todoTask = new Todo(commandArgs);
+        TASK_LIST.addTask(todoTask);
+        echo(todoTask);
+    }
+
+    private void handleDeadlineCommand(String commandArgs) throws InvalidDeadlineCommandException{
+        String[] deadlineArgParts = commandArgs.split("/by", 2);
+        if (deadlineArgParts.length != 2 || deadlineArgParts[0].isBlank()) {
+            throw new InvalidDeadlineCommandException();
+        }
+        String description = deadlineArgParts[0];
+        String deadline = deadlineArgParts[1];
+        Task deadlineTask = new Deadline(description, deadline);
+        TASK_LIST.addTask(deadlineTask);
+        echo(deadlineTask);
+    }
+
+    private void handleEventCommand(String commandArgs) throws InvalidEventCommandException{
+        String[] eventArgParts = commandArgs.split("/from", 2);
+        if (eventArgParts.length != 2 || eventArgParts[0].isBlank()){
+            throw new InvalidEventCommandException();
+        }
+        String description = eventArgParts[0];
+        String[] eventArgTimeParts = eventArgParts[1].split("/to", 2);
+        if (eventArgTimeParts.length != 2){
+            throw new InvalidEventCommandException();
+        }
+        String from = eventArgTimeParts[0];
+        String to = eventArgTimeParts[1];
+        Task eventTask = new Event(description, from, to);
+        TASK_LIST.addTask(eventTask);
+        echo(eventTask);
```

Comment:
Perhaps add blank lines to separate groups of related statements for improving readibility. Same goes for other methods with no blank spaces between several statements.